src/theory/strings/term_registry.cpp

```diff
@@ -676,9 +681,20 @@ void TermRegistry::removeProxyEqs(Node n, std::vector<Node>& unproc) const
 
 void TermRegistry::getRelevantTermSet(std::set<Node>& termSet)
 {
-  d_theory.collectAssertedTerms(termSet);
+  std::set<Node> relevantTerms;
+  d_theory.collectAssertedTerms(relevantTerms);
   // also, get the additionally relevant terms
-  d_theory.computeRelevantTerms(termSet);
+  d_theory.computeRelevantTerms(relevantTerms);
+
+  // Remove inactive terms from the relevant term set
```

Comment:
I'd rather not having this filtering builtin to this method, or make term registry depend on ExtTheory.  Actually, perhaps the combined method could be a method in ExtfSolver, which has access to both?

Reply:
Seems like a good idea. I've done that.